alu_u32/src/com/stark.rs

```diff
@@ -35,6 +35,10 @@ where
         );
         builder.assert_bool(local.not_equal);
         builder.assert_eq(local.not_equal, local.diff * local.diff_inv);
+        builder.assert_eq(
```

Comment:
This looks fine, but could optionally be shortened to
```rust
builder.assert_zero((AB::Expr::one() - local.not_equal) * local.diff);
```
Or maybe this would be the most intuitive
```rust
let equal = AB::Expr::one() - local.not_equal;
builder.when(equal).assert_zero(local.diff);
```

Reply:
Thanks!